src/CmsManager/BaseCmsPageManager.php

```diff
@@ -28,7 +28,7 @@ abstract class BaseCmsPageManager implements CmsManagerInterface
     protected $currentPage;
 
     /**
-     * @var PageBlockInterface[]
+     * @var array<PageBlockInterface|null>
```

Comment:
Isn't weird to allow null ? Shouldn't the value be filtered if needed when set.

Reply:
I was not sure about it, this property is used as a "kind of" cache:

https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/CmsManager/CmsPageManager.php#L137-L144

With that code null values get always fetched, even though we know those are null, but I changed that code to check for key exist instead (in this same PR):

https://github.com/sonata-project/SonataPageBundle/blob/14b56cf73b716587f55730d88d7130455c67be0d/src/CmsManager/CmsPageManager.php#L137-L144

So the logic now will see if the key is defined and return the null. Previously it would check on database for those nulls.

WDYT? I can change to not allow nulls too.

Reply:
Oh, I see. It can make sens then.